src/Framework/Numerical/Spline.cxx

```diff
@@ -750,7 +762,73 @@ void Spline::BuildSpline(int nentries, double x[], double y[])
   if(fInterpolator) delete fInterpolator;
 
   fInterpolator = new TSpline3("spl3", x, y, nentries, "0");
```

Comment:
The main idea is to leave everything as untouchable as possible by the following reasons:
1. for the compatibility with the previous code, because there are such methods as void Spline::SaveAsROOT, LoadFromTSpline3(), and constructor Spline (const TSpline3 &spline).
2. some functionality of TSpline3 is used in the Spline class, e.g. in the method FindClosestKnot).
3. taking into account the next comments, TSpline3 will keep all knots.

Reply:
Further, one can ask why not implement methods analogous to Spline::SaveAsROOT, LoadFromTSpline3(), and constructor Spline (const TSpline3 &spline) for ROOT::Math::Interpolator?